common/logger/src/libra_logger.rs

```diff
@@ -245,20 +245,26 @@ impl LoggerService {
         for entry in self.receiver {
             PROCESSED_STRUCT_LOG_COUNT.inc();
 
-            if let Some(writer) = &mut writer {
-                Self::write_to_logstash(writer, &entry);
-            }
-
             if let Some(printer) = &self.printer {
                 let s = format(&entry).expect("Unable to format");
                 printer.write(s)
             }
+
+            if let Some(writer) = &mut writer {
+                Self::write_to_logstash(writer, entry);
+            }
```

Comment:
interesting, I hadn't realized that these were two different things... should we actually have a Vec of printers?

Reply:
Yeah we send things out to stderr as well as to a TCP endpoint.